core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -400,6 +389,9 @@ private PromotionStatus getTopicPromotionEnv(
           if (!((envOrderList.indexOf(targetEnvId) - envOrderList.indexOf(environmentId)) == 1)
               || !envOrderList.contains(environmentId)) {
             promotionStatus.setStatus(NO_PROMOTION);
+          } else if (isTopicPromoteRequestOpen(
+              topicSearch, promotionStatus.getTargetEnvId(), tenantId)) {
+            promotionStatus.setStatus(REQUEST_OPEN);
```

Comment:
may be or not in this PR, but how about making this an enum https://github.com/aiven/klaw/blob/main/core/src/main/java/io/aiven/klaw/model/response/PromotionStatus.java#L8C8-L8C8

Reply:
110%